### PR TITLE
(#164) pass path to active configuration file to choria

### LINKED ISSUE #164: Custom configuration files aren't passed to choria during discovery
We run Choria with a number of unique configuration files, and are attempting to just now move to the most recent releases, which seem to change the discovery to shell out to `choria` vs executing directly from within MCO.

When using a custom configuration file, presently the path to that config file is not passed into Choria, and as such it tries to execute with the default.

Given this:

```
$ mco_choria_prd choria show_config
Active Choria configuration:

The active configuration used in Choria comes from using Puppet AIO defaults, querying SRV
records and reading configuration files.  The below information shows the completely resolved
configuration that will be used when running MCollective commands

MCollective related:

    MCollective Version: 2.24.1
         Choria Version: 0.19.0
     Client Config File: /Users/chris/.choriarc
     Active Config File: /Users/chris/.mcollective.choria.d/prd/client.conf
```

I'd expect: `choria discover -j --silent --config /Users/chris/.mcollective.choria.d/prd/client.conf -T mcollective -A rpcutil --dm mc` but instead, the following is executed: `choria discover -j --silent -T mcollective -A rpcutil --dm mc`

### PR DESCRIPTION
**Pull request by chrisboulton**
Resolves #164

### CODE REVIEW

**Review by BC-Dat — APPROVED** [review]

File: lib/mcollective/discovery/delegate.rb [hunk 1]
```diff
@@ -10,6 +10,7 @@ def self.discover(filter, timeout, limit, client)
 
         cmd = [binary_name, "discover", "-j", "--silent"]
 
+        cmd << "--config" << client.options.fetch(:config)
```
Comment (ripienaar) [inline lib/mcollective/discovery/delegate.rb:13]: just to be safe lets do: 

```
config = client.options.fetch(:config)
cmd << "--config" << config if config
```

Comment (chrisboulton) [reply lib/mcollective/discovery/delegate.rb]: @ripienaar Yup - updated. I did trace the code and `:config` seems to always be set, but better to be defensive & safe. 👍

Comment (ripienaar) [reply lib/mcollective/discovery/delegate.rb]: Yeah it really should be always set :)

Will take a look soon

**Comment by ripienaar** [comment]
How urgent is this for yous? Do you have the ability to carry a local patch? Unfortunately you filed this like 2 days after I did a release hehe so just need to know if I should ship this for you

**Comment by chrisboulton** [comment]
@ripienaar We can wait - most of our folks are using the older support libs that don't wrap `choria`. The rest of us can just patch locally for now. 👍

**Comment by ripienaar** [comment]
Cool, maybe of interest (maybe not) but you can now have a per directory/project configuration by placing `choria.conf` in a directory.  This will merge with overrides the system default config.

Might be an alternative approach to consider.

**Comment by chrisboulton** [comment]
Yup - that definitely is. I'll have a look later -- very interested in undoing some of this more bespoke stuff we've got.

**Pull request merged**